internal/cli/app_docs.go

```diff
@@ -270,6 +272,55 @@ func (c *AppDocsCommand) emitSection(w io.Writer, name, use, h string, fields []
 	}
 }
 
+// jsonFormat attempts to output all the data included in a a plugin's Documentation() function in the JSON file format
+func (c *AppDocsCommand) jsonFormat(name, ct string, doc *docs.Documentation) {
+	// we use this constant to compare to ct for some special behavior
+	const csType = "configsourcer"
+
+	w, err := os.Create(fmt.Sprintf("./docs/%s-%s.json", ct, name))
+	if err != nil {
+		panic(err)
+	}
+
+	jMap := map[string]interface{}{"name": name, "type": ct}
```

Comment:
⚪️ This section of code (L280 - L309) seems to duplicate the same section of the `jsonFormatConfigSourcer` function (L398 - L420). Instead of using the switch statement in `app_docs.go` to determine which function to use, could we have just one function which has different behavior within it based on the plugin type?

It might be worthwhile to create helper functions within it if necessary, but this could help with DRYness. 😄